examples/api/lib/cupertino/sheet/cupertino_sheet.1.dart

```diff
@@ -0,0 +1,173 @@
+  }
+}
+
+class HomePage extends StatelessWidget {
+  const HomePage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      navigationBar: const CupertinoNavigationBar(
+        middle: Text('Sheet Example'),
+        automaticBackgroundVisibility: false,
+      ),
+      child: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            CupertinoButton.filled(
+              onPressed: () {
+                Navigator.of(context).push(CupertinoSheetRoute<void>(
+                  pageBuilder: (BuildContext context) => const SheetContentWithNavigator()
+                ));
+              },
+              child: const Text('Open Bottom Sheet'),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class SheetContentWithNavigator extends StatelessWidget {
+  const SheetContentWithNavigator({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final BuildContext parentContext = context;
+    return Navigator(
+      initialRoute: '/',
+      onGenerateRoute: (RouteSettings settings) {
+        return CupertinoPageRoute<void>(
+          builder: (BuildContext context) {
+            return PopScope(
+              canPop: settings.name != '/',
+              onPopInvokedWithResult: (bool didPop, Object? result) {
+                if (didPop) {
+                  return;
+                }
+                Navigator.of(parentContext).pop();
+              },
+              child: Semantics(
+                scopesRoute: true,
+                explicitChildNodes: true,
+                child: const SheetScaffold(),
+              ),
+            );
+          }
+        );
+      },
+    );
+  }
+}
+
+class SheetScaffold extends StatelessWidget {
+  const SheetScaffold({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final BuildContext? topLevelContext = CupertinoSheetController.maybeOf(context)?.topLevelContext;
+    return CupertinoPageScaffold(
+      child: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            const Text('The sheet'),
+            CupertinoButton.filled(
+              onPressed: () {
+                Navigator.of(context).maybePop();
+              },
+              child: const Text('Go back'),
+            ),
+            CupertinoButton.filled(
+              onPressed: () {
+                Navigator.of(context).push(
+                  CupertinoPageRoute<void>(builder: (BuildContext context) => const SheetNextPage())
+                );
+              },
+              child: const Text('Push Page'),
+            ),
+            CupertinoButton.filled(
+              onPressed: () {
+                Navigator.of(topLevelContext ?? context).push(CupertinoSheetRoute<void>(
+                  pageBuilder: (BuildContext context) => const SheetContentWithNavigator()
```

Comment:
Because of how odd this is, rather than having people use the `CupertinoSheetRoute`, it's probably better to do something like `showCupertinoSheet`.

Reply:
I mentioned it elsewhere but I agree that the showCupertinoSheet is probably worth it for convenience. Though what's your concern with the complexity here, is it the need to use CupertinoSheetRoute and to get the BuildContext right? Compare that with a normal usage of `Navigator.of(context).push` like you have below:

```dart
Navigator.of(context).push(
  CupertinoPageRoute<void>(builder: (BuildContext context) => const SheetNextPage())
);
```